Add support for git submodules. Addresses issue #54.

cojomojo opened the pull request.

mixu commented on lib/command-requirements.js:
```diff
@@ -5,12 +5,20 @@ module.exports = {
+      else {
+        // check if the directory is a submodule
+        // addresses https://github.com/mixu/gr/issues/54
+        var path = req.path.split(/(\\|\/)/).slice(0, -2).join('/');
```
`path.dirname(path.dirname(req.path))`?

mixu commented on lib/command-requirements.js:
```diff
@@ -5,12 +5,20 @@ module.exports = {
+        stat = fs.statSync(path + '/.git/');
+        if (stat.isDirectory()) {
+          stat = fs.statSync(path + '/.gitmodules')
+          if (stat.isFile()) return true;
```
nit: needs curly braces

cojomojo commented:
@mixu I updated the code according to your suggestions, and moved it into the catch block since statSync *should* throw an error when trying to access a non-existent directory. I say should because of https://github.com/nodejs/node/issues/9106.

mixu commented:
:+1: thanks for making these changes and for the PR! Merged!

mixu commented:
published as `v0.5.2`

cojomojo commented:
@mixu you're welcome, thanks for the great tool!
